maintenance/wikia/fixPageLatest.php

```diff
@@ -45,44 +74,49 @@ public function execute() {
 		);
 
 		$count = $res->numRows();
-		$this->output(" {$count} pages(s) found\n\n");
+		$this->output(" {$count} affected pages(s) found\n\n");
 
 		if ($count === 0) {
 			$this->output("No articles found!\n");
-			die();
+			die(1);
 		}
 
 		// fix entries
 		$fixed = 0;
-		$dbw = $this->getDB(DB_MASTER);
+
+		$this->dbw = $this->getDB(DB_MASTER);
+		$this->user = User::newFromName( 'WikiaBot' );
 
 		while($row = $res->fetchObject()) {
 			$revId = intval($row->rev_id);
 
-			// no revision data
-			if ($revId === 0) {
+			if ( $isDryRun ) {
+				$this->output("* {$row->page_title} affected - would set page_latest to {$revId}\n");
 				continue;
 			}
 
 			$this->output("* {$row->page_title}");
 
-			if ($isDryRun) {
-				$this->output(" - page_latest would be set to {$revId}\n");
-			}
-			else {
-				$dbw->update(
-					'page',
-					['page_latest' => $revId],
-					['page_id' => $row->page_id],
-					__METHOD__
-				);
-				$this->output(" - page_latest set to {$revId}\n");
-
-				$fixed++;
+			// no revision data - we can generate an empty revision if "page_len" is set to 0 (PLATFORM-1286)
+			if ($revId == 0 &&  $row->page_len == 0 ) {
+				$this->output(" - making an empty edit");
+				$revId = $this->insertEmptyRevision( $row->page_id );
```

Comment:
Shouldn't we check if database INSERT was successful? Will it throw exception on failure and if so should we catch it here?